lib/beaker/dsl/install_utils/foss_utils.rb

```diff
@@ -1097,98 +1054,21 @@ def install_puppet_agent_dev_repo_on( hosts, opts )
             opts[:puppet_collection] ||= 'PC1'
             add_role(host, 'aio') #we are installing agent, so we want aio role
             release_path = opts[:download_url]
-            variant, version, arch, codename = host['platform'].to_array
             copy_dir_local = File.join(opts[:copy_base_local], variant)
             onhost_copy_base = opts[:copy_dir_external]
 
             case variant
-            when /^eos/
-              release_path_end, release_file = host.get_puppet_agent_package_info(
-                opts[:puppet_collection], opts[:puppet_agent_version] )
+            when /^(fedora|el|centos|debian|ubuntu|cumulus)$/
+              sha = opts[:puppet_agent_sha] || opts[:puppet_agent_version]
+              install_puppetlabs_dev_repo( host, 'puppet-agent', sha)
```

Comment:
This seems to do the trick:

```ruby
install_puppetlabs_dev_repo( host, 'puppet-agent', sha, opts[:copy_base_local], opts )
```

Using `copy_base_local` instead of `copy_dir_local`, because using the latter causes `variant` to be added twice, e.g. `tmp/repo_configs/debian/debian/wheezy`